File: SDK/Models/Messages/BrokerMessage.cs
namespace Agience.SDK.Models.Messages
{
    public enum BrokerMessageType
    {
        EVENT,
        INFORMATION,
        CONTEXT,
        UNKNOWN
    }

    public class BrokerMessage
    {
        public BrokerMessageType Type { get; set; } = BrokerMessageType.UNKNOWN;
        public string? Topic { get; set; }
        public string? SenderId => Topic?.Split('/')[0];
        public string? Destination => Topic?.Substring(Topic.IndexOf('/') + 1);
        //public string? Payload { get; set; }

        private object? _content;

        public Data? Data
        {
            get => Type == BrokerMessageType.EVENT ? (Data?)_content : null;
            set
            {
                if (Type == BrokerMessageType.EVENT)
                {
                    _content = value;
                    //Payload = value?.Raw;
                }
            }
        }

        public Information? Information
        {
            get => Type == BrokerMessageType.INFORMATION ? (Information?)_content : null;
            set
            {
                if (Type == BrokerMessageType.INFORMATION)
                {
                    _content = value;
                    //Payload = value != null ? JsonSerializer.Serialize(value) : null;
                }
            }
        }
    }
}

File: python/src/core/models/messages/broker_message.py
from enum import Enum
from typing import Optional, Union, Any
from pydantic import BaseModel, Field, computed_field, PrivateAttr
from core.data import Data
from core.information import Information


class BrokerMessageType(Enum):
    EVENT = "EVENT"
    INFORMATION = "INFORMATION"
    CONTEXT = "CONTEXT"
    UNKNOWN = "UNKNOWN"


class BrokerMessage(BaseModel):
    type: BrokerMessageType = Field(default=BrokerMessageType.UNKNOWN)
    topic: Optional[str] = None

    data: Optional[Data] = None
    information: Optional[Information] = None

    def model_post_init(self, __context: Any) -> None:
        if self.type == BrokerMessageType.EVENT:
            self.information = None
        elif self.type == BrokerMessageType.INFORMATION:
            self.data = None

    @computed_field
    def sender_id(self) -> Optional[str]:
        if self.topic:
            parts = self.topic.split('/')
            if len(parts) > 1:
                return parts[1]
        return None

    @computed_field
    def destination(self) -> Optional[str]:
        if self.topic:
            parts = self.topic.split('/')
            if len(parts) > 2:
                return "/".join(parts[2:])
        return None
